Review

xavierhd opened the pull request.

xavierhd commented on src/main.vala:
```diff
@@ -0,0 +1,9 @@
+int main (string[] args) {
+    if (Thread.supported () == false) {
+        stderr.printf ("Threads are not supported!\n");
```
Since this is a GUI app, this message should be displayed in a message box.

Vysp3r replied:
Le truc c'est que sa viens de la documentation officiel et je sais aucunement si la lib j'utilises permet de ne pas avoir de support pour sa. Et oui c'est un app GUI, mais beaucoup d'app linux utilise la console pour output leur erreurs. Quand une app j'utilises bug je la lance avec le terminal justement pour voir ca. J'ai pas d'ordi pour tester sa tristement. Une fois cette verification de faites je vais faire juste des messages box avec des messages simples et le full exception dans le terminal, mais pour cette partie je sais pas si je peux vrm afficher un truc si ya pas thread. Je vais regarder voir si je peux faire une VM qui correspond a ca pour pouvoir tester.

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+namespace ProtonPlus.Manager {
+    public class File {
+        public static void Extract (string install_location, string launcher_name, string tool_name) {
```
Lowercase method name.

Vysp3r replied:
Yeah je sais jai pris une mauvaise habitude avec ca :/

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+            ext.set_standard_lookup ();
+            ext.set_options (flags);
+
+            if (archive.open_filename (install_location + tool_name + ".tar.gz", 1920000) != Archive.Result.OK) return;
```
Magic number 1920000... maybe put it in a meaningful CONST?

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+
+            unowned Archive.Entry entry;
+
+            int bob = 0;
```
👀 I know that bob might be a zero, but what is he doing here?

Vysp3r replied:
Ta raison ya aucun rapport la!

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+                if (r == Archive.Result.EOF) break;
+                if (r < Archive.Result.WARN) break;
+                entry.set_pathname (install_location + entry.pathname ());
+                if(bob++ == 0){
```
The condition is only true on the first iteration. A boolean flag would be clearer.

Vysp3r replied:
C'etait pour tester, meme chose pour les variables bob et test.

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+                if (r < Archive.Result.WARN) break;
+                entry.set_pathname (install_location + entry.pathname ());
+                if(bob++ == 0){
+                    test = entry.pathname ();
```
Var `test` should be named `pathname`.

Vysp3r replied:
J'ai juste forget de remove la variable test :)

xavierhd commented on src/manager/file.vala:
```diff
@@ -0,0 +1,71 @@
+                if(bob++ == 0){
+                    test = entry.pathname ();
+                }
+                r = ext.write_header (entry);
```
Why don't you check the content of `r` here? This `r` value is never used.

Vysp3r replied:
r c'est juste pour stocker le resultat des operations. Elle est utiliser si tu regardes bien dans les if pour verifier il y a eu un warning ou si la fin du fichier (EOF) a ete atteint.

xavierhd replied:
Ouai, mais ce que tu assignes à la ligne 35 n'est jamais utilisé. Ajoute un if pour checker la valeur ou enlève l'assignation?

Vysp3r replied:
Ah ok, j'avais enlever des messages d'erreurs, mais je les ai remis. La, la valeur est utiliser par apres. Merci :)

xavierhd commented on src/manager/themes.vala:
```diff
@@ -0,0 +1,11 @@
+namespace ProtonPlus.Manager {
+    public class Themes {
+        public static void Load() {
```
Lowercase

Vysp3r replied:
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#pascal-case

xavierhd reviewed (commented):
This is the part 1, I'm falling asleep so the rest will wait for another day.

xavierhd commented:
@Vysp3r , j'ai commencé un review ici, mais je vais le finir une autre fois, je suis trop fatigué pour continuer :D

Vysp3r commented:
@xavierhd oh nice merci beaucoup! je suis en train de faire un gros refactor en plus, sa tombe pile poile :)

Vysp3r commented:
Entouka quand tu va rechecker sa va avoir changer pas mal ahah. Un site de linux que je suis beaucoup a faites [un post sur mon application](https://www.gamingonlinux.com/2022/12/protonplus-is-a-new-in-development-proton-version-manager/) pis sa la boost mes stars et le traffique de mon repo. Sa ma boost et me pousse encore plus, je suis vraiment content :)

Vysp3r committed:
:memo: Update README.md

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:hammer: Update meson

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:bookmark: Merge dev into main (v0.3.1)

Commit:
add indonesian translation

dreamyukii committed:
fix something

Commit:
Add Brazilian Portuguese(pt-BR) translation

Vysp3r committed:
:lipstick: Lock dialogs to main window

Vysp3r committed:
Merge branch 'dev' into brazilian-portuguese-not-spanish

Vysp3r committed:
Add Brazilian Portuguese (pt-BR) translation

Vysp3r committed:
Merge branch 'dev' into main

Signed-off-by: Charles Malouin <62019735+Vysp3r@users.noreply.github.com>

Vysp3r committed:
add Indonesian language

wehagy committed:
:memo: Fix README.md typo

Signed-off-by: Wesley Hamada Gimenes <95216845+wehagy@users.noreply.github.com>

wehagy committed:
:memo: Add desktop-file-utils link to README.md

Signed-off-by: Wesley Hamada Gimenes <95216845+wehagy@users.noreply.github.com>

Vysp3r committed:
Merge branch 'dev' into throwing-stars

Vysp3r committed:
:memo: Fix README.md typo

Vysp3r committed:
:bug: Fix possible crash scenario

Vysp3r committed:
:bug: Fix possible crash scenario

Vysp3r committed:
:technologist: Project refactor

Vysp3r committed:
:bug: Fix tool renaming

wehagy committed:
:hammer: flatpak build from current dir

Signed-off-by: Wesley H Gimenes <95216845+wehagy@users.noreply.github.com>

Commit:
:memo: Rework and add instructions how to build with flatpak

Vysp3r committed:
:bug: Fix download not working on some machine

Vysp3r committed:
:bug: Fix tool renaming

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:bookmark: Merge dev into main (v0.3.2)

albanobattistella committed:
Create it.po

Signed-off-by: albanobattistella <34811668+albanobattistella@users.noreply.github.com>

albanobattistella committed:
Update LINGUAS

Signed-off-by: albanobattistella <34811668+albanobattistella@users.noreply.github.com>

albanobattistella committed:
Rename it.po to it_IT.po

Signed-off-by: albanobattistella <34811668+albanobattistella@users.noreply.github.com>

Vysp3r committed:
Correct desktop file exec

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
Update README.md

Signed-off-by: Charles Malouin <62019735+Vysp3r@users.noreply.github.com>

Vysp3r committed:
Merge branch 'dev' into main

Vysp3r committed:
Create Italian Translation

Vysp3r committed:
:memo: Update README.md

Signed-off-by: Charles Malouin <62019735+Vysp3r@users.noreply.github.com>

Vysp3r committed:
Merge branch 'dev' into flatpak-the-builder

Commit:
Add Chinese(Simplified) Translation

Diego-Ivan committed:
Fix Translations:

* Remove specific locale codes to per language ones.
* In Application.vala, bind domain in construction.

Vysp3r committed:
Merge branch 'dev' into wip/diegoivan/fix-po

Signed-off-by: Charles Malouin <62019735+Vysp3r@users.noreply.github.com>

Vysp3r committed:
Fix Translations

Vysp3r committed:
Merge branch 'dev' into dev

Signed-off-by: Charles Malouin <62019735+Vysp3r@users.noreply.github.com>

Vysp3r committed:
Add Chinese(Simplified) Translation

Vysp3r committed:
Merge branch 'dev' into flatpak-the-builder

Vysp3r committed:
:memo: Add how to build from source with flatpak

Vysp3r committed:
:bug: Fix translation

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:bookmark: Merge dev into main (v0.3.4)

Vysp3r committed:
:globe_with_meridians: Add missing translations

smotim committed:
Translated a few missing words

Signed-off-by: Timofey Smolin <57951811+smotim@users.noreply.github.com>

Vysp3r committed:
Translated a few missing words

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:bookmark: Merge dev into main (v0.3.5)

Vysp3r committed:
Update com.vysp3r.ProtonPlus.appdata.xml

Vysp3r committed:
:bug: Fix directory creation

Vysp3r committed:
:bug: Fix Flatpak permissions

Vysp3r committed:
:sparkles: Added Proton Tkg

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:sparkles: Add warning for gamescope compatibility issues

Vysp3r committed:
:mute: Remove debug log

Vysp3r committed:
:bug: Fix download getting cancelled randomly

Vysp3r committed:
:globe_with_meridians: Added missing translations

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

smotim committed:
added a missed ru translations

smotim committed:
Update ru.po

Signed-off-by: Timofey Smolin <57951811+smotim@users.noreply.github.com>

Vysp3r committed:
Merge branch 'main' into dev2

Vysp3r committed:
added a missed ru translations

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
Fixed installer tool switching

In the installer, when you were switching tool, it was not always working especially with Proton Tkg. Now it should be fixed.

Vysp3r committed:
:mute: Remove debug log

Vysp3r committed:
:lipstick: Updated icons

I changed the icons to be more user-friendly.

Vysp3r committed:
:bug: Fixed tools and releases not being locked when needed

When installing a tool or switching tool, you will not be able to switch tool or release anymore to prevent problems caused by that until the action is completed.

Vysp3r committed:
:bug: Fixed progress bar not being hidden when needed

After installing a tool, if you switch tool or release, the progress bar will now hide itself until you download a new tool.

Vysp3r committed:
:globe_with_meridians: Updated translation files

Nothing that would require translation from the community is required here.

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:sparkles: Updated the About window

I added a row to show the space of the selected tool/launcher. I modified the row named Directory to Path since in my opinion it was better it was a better word for it. I updated the translation files since new text was added. Note that the new GetDirectorySize function is using a process to get the size with the 'du -sh $path' command. It's something I never used and I don't know how well it will work with Flatpak.

Vysp3r committed:
Updated the About window

I added a row to show the space of the selected tool/launcher. I modified the row named Directory to Path since in my opinion it was better it was a better word for it. I updated the translation files since new text was added. Note that the new GetDirectorySize function is using a process to get the size with the 'du -sh $path' command. It's something I never used and I don't know how well it will work with Flatpak.

Vysp3r committed:
Merge branch 'main' of https://github.com/Vysp3r/ProtonPlus

windblows95 committed:
Added utilities for directories

windblows95 committed:
Updated GetDirectorySize function

Made it use stat to get folder size instead of spawning du

windblows95 committed:
Replaced the Delete function

Old Delete function moved directories to trash

windblows95 committed:
Fixed formatting of folder size

Vysp3r committed:
Added utilities for folders and files

windblows95 committed:
Moved the Clean launcher function to an other thread

Vysp3r committed:
Migrated preferences system to GLib.Settings

Instead of using my own system to save the preferences, I opted to use GLib.Settings which will make it easier for me in the future to add stuff since the way I was doing it was ugly and hard to maintain.

Vysp3r committed:
Moved the Clean launcher function to an other thread

Vysp3r committed:
Merge branch 'main' into new-settings-system

Vysp3r committed:
Migrated preferences system to GLib.Settings

windblows95 committed:
Implementated cancelation dialog when downloading

windblows95 committed:
Increased range of filesystem access to Lutris, Heroic and Bottles folders

windblows95 committed:
Updated language files

Vysp3r committed:
Implementated cancelation dialog when downloading & modified filesystem permissions scopes

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
Update SECURITY.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:sparkles: Added support for Heroic Games Launcher

Since it was broken and never really worked, I removed it a long time ago, but now it's back and working like it should!

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
📝 Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:sparkles: Added support for Bottles

Since Bottles uses a custom list for there tool releases, adding tools version that are not supported by Bottles will make that tool unusable by Bottles. Sadly, I cannot do much about it for now, but at least you can kinda manage Bottles now. I've verified what ProtonUp-Qt do and they do not have a solution for that right now. I will probably need to host a server which will serve a json with all their versions and keep it up to date manually so that everyone using ProtonPlus have the best experience with Bottles. I'm looking foward to making such a system.

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:speech_balloon: Update the release notes

Vysp3r committed:
:memo: Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Added better support for gamescope

ProtonPlus will now start in fullscreen when running with gamescope

Vysp3r committed:
Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:construction: WIP

Added most of the stuff of the new UI, but still needs some few things before going live.

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
:construction: WIP

Vysp3r committed:
✨ Complete UI Overhaul -  🧑‍💻 Project Refactor (#91)

Vysp3r committed:
:memo: Updated Flathub download logo

It's now using the official download badge instead of a random one from the internet. I've also updated the alt tag and the width to make it bigger.

Vysp3r committed:
:memo: Updated Flathub download logo (#93)

Vysp3r committed:
:bug: Fix Steam install  message

The message that showed up at the end of each install asking to restart Steam if it was open will now only being displayed for Steam instead of every launcher. I updated the translation files at the same time.

Vysp3r committed:
:bug: Fix Steam install message (#94)

Vysp3r committed:
:bug: Fix preferences page being empty

When the user didn't have any available setting, it was showing an empty page. I added something so that it does not feel empty, because of a bug. I updated the translation files at the same time since I added some text.

Vysp3r committed:
:bug: Fix preferences page being empty (#95)

Vysp3r committed:
:bug: Fix window always being able to be closed

I added a check so that when you're installing a tool, you cannot close the window until you cancel or wait for the installation to be done.

Vysp3r committed:
:bug: Fix window always being able to be closed (#96)

Vysp3r committed:
:lipstick: Updated the installer UI

I decided to remove the back button that was previously there and just update the cancel button to a go back button. I updated the translation files at the same time since I added new texts.

Vysp3r committed:
:lipstick: Updated the installer UI (#97)

Vysp3r committed:
:sparkles: Added a tasks queue system

I added a tasks queue system so that when deleting a tool, if you quit before it's done, it still finish that task before closing  the application. The application will be hidden if there's some tasks left to do and you close it.

Vysp3r committed:
:sparkles: Added a tasks queue system (#98)

Vysp3r committed:
:hammer: Update appdata

I updated the appdata file to add all the commits for the v0.4.1 update. I updated the application version in the meson. I updated the translation files in case I forgot to update them on some new texts.

Vysp3r committed:
:hammer: Update appdata (#99)

Vysp3r committed:
:memo: Change preview video file format to webm

Vysp3r committed:
:memo: Change preview video file format to webm (#102)

AsciiWolf committed:
Fix typo

AsciiWolf committed:
Add Czech translation

Vysp3r committed:
Fix typo, Add Czech translation (#103)

Vysp3r committed:
:art: Improved overall code structure

I made sure public properties where starting with a capitalized letter. When a  window needed something from another window, I made sure to pass the window as a parameter instead of passing every specific thing I needed. I moved the launcher list to the top window so that we can easily access it in any window. I also cleaned some useless lines.

Vysp3r committed:
:art: Improved Utils.Filesystem code structure

I combined Utils.File and Utils.Dir into Utils.Filesystem. I renamed some functions with a better name. I cleaned a bit the code at the same time.

Vysp3r committed:
:fire: Remove .vscode folder

This folder was added by accident and should not be there.

Vysp3r committed:
:bug: Fixed release deletion not always working

The delete function was not using the correct name for the directory which was causing some tool to not get deleted.

ChrisLauinger77 committed:
Add german translation

Vysp3r committed:
:lipstick: Added missing arrow

I added an arrow next to each available tools to make it easier to the user to understand that you can click on it.

Vysp3r committed:
:sparkles: Added caching system

I added a basic caching system so that in case you're not connected to the internet you can still browse places where you already loaded data. This is just the first step, I plan on making this system better in the future, but for now it will do.

Vysp3r committed:
Added german translation (#110)

Vysp3r committed:
Backend optimization (#111)

Vysp3r committed:
:wrench: Update appdata and meson

I updated the appdata to include the stuff needed for 0.4.2 and updated the meson version to 0.4.2.

Vysp3r committed:
:globe_with_meridians: Update language files

I updated the language files for the new text added in 0.4.2.

Vysp3r committed:
:lipstick: Updated the UI

Vysp3r committed:
:memo: Fix appdata typo

Vysp3r committed:
:lipstick: New UI and optimization (#115)

Vysp3r committed:
:bug: Fix warning box

I I added an automatic check so that if you have no launcher installed, you can simply install one and the UI will refresh so you don't have to restart the application. I also updated the translation files since I forgot to do it last time I pushed.

Vysp3r committed:
:bug: Fix warning box and translation (#116)

ChrisLauinger77 committed:
Update  german translation

Vysp3r committed:
Update german translation (#118)

Vysp3r committed:
:bug: Fix install directories creation

I used the wrong path to check if it existed before so it was not creating the directories when opening the application which would then cause problems when installing a runner.

Vysp3r committed:
:bug: Fix install directories creation (#120)

Vysp3r committed:
:bug: Fix Proton Tkg

Vysp3r committed:
:bug: Fix Proton Tkg (#121)

Vysp3r committed:
:hammer: Update appdata

Vysp3r committed:
:memo: Update the README

Vysp3r committed:
:bento: Updated the preview image

Vysp3r committed:
:hammer: Updated the appdata

Vysp3r committed:
Updated README, appdata and preview image (#123)

Vysp3r committed:
Update README.md

Since the application is pretty stable, there's no need to say it's a WIP anymore.

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Update README.md (#124)

Vysp3r committed:
:bug: Fix Flathub download link

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:bug: Fix Flathub download link (#125)

Vysp3r committed:
:memo: Update README

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:memo: Update README (#126)

Vysp3r committed:
:memo: Update README

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
:memo: Update README (#127)

Vysp3r committed:
:recycle: Update delete dialog

The dialog now ask to confirm you want to delete the tool instead of doing it automatically after the dialog was dismissed.

Vysp3r committed:
:recycle: Update delete dialog (#131)

Vysp3r committed:
:sparkles: Added the ability to download older versions

Since someone requested it and it made sense to add this feature I decided to just add it.

Vysp3r committed:
:bug: Fixed caching system

Due to the new way to load previous releases, it broke the caching system so this fix is just to make it work again.

Vysp3r committed:
:sparkles: Added the ability to download older versions (#132)

Vysp3r committed:
:globe_with_meridians: Updated localization

Since I added a new option to load more releases I need to update those. At the same time I updated the text of that new feature to be shorter and less redundant.

Vysp3r committed:
:globe_with_meridians: Updated localization (#133)

Vysp3r committed:
:hammer: Updated appdata and meson

ChrisLauinger77 committed:
Update  german translation

Vysp3r committed:
Update  german translation (#136)

Vysp3r committed:
💄 Improved Steam Deck experience

I made it so that the hamburger menu changes the content of the window instead of opening a popup since that does not work in gaming mode.

Vysp3r committed:
💄 Improved Steam Deck experience (#138)

Vysp3r committed:
:globe_with_meridians: Updated localization

Vysp3r committed:
:globe_with_meridians: Updated localization (#139)

ChrisLauinger77 committed:
Update  german translation

Vysp3r committed:
Update  german translation (#140)

Vysp3r committed:
:sparkles: Added DXVK GPL+Async (Ph42oN)

This new runner required to implement a GitLab backend to be able to get data from that new source.

Vysp3r committed:
:lipstick: Added description to groups

I decided to add description to groups so that I can explain what that group is about and to try to make thing less repetitive.

Vysp3r committed:
:sparkles: Added Kron4ek Wine-Builds Staging

I added Kron4ek Wine-Builds Staging to Lutris and also fixed invalid releases showing up in Kron4ek runners. I did also update all the description from the Kron4ek runners and removed the repetitive text 'Use this when you don't know what to choose' from every runner description.

Vysp3r committed:
:sparkles: Added VKD3D support for Lutris

The 'VKD3D-Lutris' uses the same naming scheme that Lutris uses so that previously downloaded releases show up in the list.

Vysp3r committed:
:sparkles: Added Kron4ek builds and Proton-tkg to HGL

Vysp3r committed:
General improvements (#141)

Vysp3r committed:
:sparkles: Added an installed only filter

Vysp3r committed:
:bug: Fix filter switching

Vysp3r committed:
:sparkles: Add an installed only filter (#143)

Vysp3r committed:
:globe_with_meridians: Updated localization

I also removed two debug message that I left by accident and I updated the appdata.

Vysp3r committed:
:hammer: Update appdata

albanobattistella committed:
Update it.po

Signed-off-by: albanobattistella <34811668+albanobattistella@users.noreply.github.com>

Vysp3r committed:
Update it.po (#145)

Vysp3r committed:
🌐 Updated localization

ChrisLauinger77 committed:
Updated german translation

Vysp3r committed:
Updated german translation (#146)

Vysp3r committed:
:memo: Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Translated using Weblate (French)

Currently translated at 6.2% (2 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/fr/

Vysp3r committed:
Translated using Weblate (French)

Currently translated at 9.3% (3 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/fr/

Vysp3r committed:
Translations update from Hosted Weblate (#149)

Vysp3r committed:
Translated using Weblate (French)

Currently translated at 12.5% (4 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/fr/

Vysp3r committed:
Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Update README.md

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Translations update from Hosted Weblate (#150)

Commit:
Translated using Weblate (French)

Currently translated at 21.8% (7 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/fr/

Vysp3r committed:
Translations update from Hosted Weblate (#151)

Vysp3r committed:
:lipstick: Modernised the UI

I modernized the user interface using the latest Adwaita widgets. I also updated the runtime version of the application to GNOME 45 to be able to use the latest Gtk/Adwaita widgets. I also  reorganized the files. I also made so that the desktop file and the appdata file are now translatable.

Vysp3r committed:
:memo: Fix logo in README

I changed the location of the logo in a previous commit, but forgot to update the path in the README.

rezaalmanda committed:
Translated using Weblate (Indonesian)

Currently translated at 40.6% (13 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/id/

Vysp3r committed:
🧑‍💻 Backend refactor

gallegonovato committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/es/

RouHim committed:
Translated using Weblate (German)

Currently translated at 100.0% (32 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/de/

rokam committed:
Translated using Weblate (Portuguese)

Currently translated at 56.2% (18 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/pt/

rezaalmanda committed:
Translated using Weblate (Indonesian)

Currently translated at 53.1% (17 of 32 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/id/

Vysp3r committed:
🧑‍💻 Backend refactor

Vysp3r committed:
:memo: Update README.md

Vysp3r committed:
🌐 Updated localization

Vysp3r committed:
Update preview image

Vysp3r committed:
ProtonPlus Rework (#153)

Vysp3r committed:
Merge branch 'main' into weblate-protonplus-protonplus

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
Translations update from Hosted Weblate (#152)

Vysp3r committed:
Fix warning

Vysp3r committed:
🌐 Updated localization

Vysp3r committed:
Merge branch 'main' into fix-warning-loc

Signed-off-by: Charles Malouin <dev@vysp3r.com>

Vysp3r committed:
🌐 Updated localization

Vysp3r committed:
Fix a warning and update localization (#154)

Vysp3r committed:
Update meson.build

Vysp3r committed:
Update com.vysp3r.ProtonPlus.appdata.xml.in

gallegonovato committed:
Translated using Weblate (Spanish)

Currently translated at 95.2% (40 of 42 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/es/

gallegonovato committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (42 of 42 strings)

Translation: ProtonPlus/ProtonPlus
Translate-URL: https://hosted.weblate.org/projects/protonplus/protonplus/es/

ChrisLauinger77 committed:
Update german translation

Vysp3r committed:
Translations update from Hosted Weblate (#156)

Vysp3r committed:
Update german translation (#157)

Vysp3r committed:
:bug: Fix Bottles icon missing

The path to the icon was invalid for the system version of Bottles so I just fixed the path.

Vysp3r committed:
:bug: Fix Bottles icon missing (#159)

Vysp3r committed:
Update appdata
